### PR TITLE
Add `libpak-tools package bundle` command which builds and packages a buildpack

### PR DESCRIPTION
**Pull request by dmikusa**
## Summary

This command runs the build workflow for libcnb & libpak based buildpacks. It both compiles (`libpak-tools package compile`) and also packages the compiled buildpack into a buildpack image which is loaded into the Docker daemon.

It does not presently work for non-libcnb/libpak buildpacks because they have a slightly different build system, and specifically the way that the version number is set differs. This can be worked around, but requires a separate PR to libcnb to update the compilation command and how it handles setting the version number.

This tool is build off libpak v2, but should work with libpak v1 buildpacks as well.

## Use Cases

An easy way to compile & package libcnb & libpak buildpacks.

## Checklist

* [ ] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [ ] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by dmikusa** [comment]
Rebased.

I'm a little unsure about these `go.work` and `go.work.sum` files. That is something VScode suggested I do, but it didn't seem to cause issue if I did not do that either. I think it's necessary because of the `tools/` sub-directory which has tools for running the linter and building. Maybe there's a better way to do this all? They are really just dev dependencies, but I'm not 100% sure how to do that in Go.

If you think I should look into this more before we merge, let me know.

**Comment by pivotal-david-osullivan** [comment]
I had a look at these files and I read they're not intended to be checked in? https://github.com/golang/go/issues/51941#issuecomment-1941970747

**Comment by dmikusa** [comment]
@pivotal-david-osullivan Interesting. OK, removed. If it becomes annoying, we can add them later.

File: packager/buildpack.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+func NewBundleBuildpackForTests(executor effect.Executor, exitHandler libcnb.ExitHandler) BundleBuildpack {
+	return BundleBuildpack{
+		executor:    executor,
+		exitHandler: exitHandler,
+	}
+}
+
+// InferBuildpackPath infers the buildpack path from the buildpack id
+func (p *BundleBuildpack) InferBuildpackPath() error {
```
Comment (anthonydahanne) [inline packager/buildpack.go:72]: my compiler complains a bit cause you use a mix of pointer and value receiver methods on `BundleBuildpack`

WDYT?

File: packager/buildpack.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+		})
+		if err != nil {
+			return fmt.Errorf("unable to execute `docker image rm` command\n%w", err)
+		}
+	}
+
+	return nil
+}
+
+// Execute runs the package buildpack command
```
Comment (anthonydahanne) [inline packager/buildpack.go:169]:
```suggestion
// ExecutePackage runs the package buildpack command
```

**Review by anthonydahanne — CHANGES_REQUESTED** [review]

**Review by pivotal-david-osullivan — APPROVED** [review]

File: packager/buildpack.go (hunk 1 above)
Comment (dmikusa) [reply packager/buildpack.go:72]: I don't get any warnings, but I think I know what you're talking about and I just pushed up a change to make them all pointers (cause we need pointer for one of the methods).

**Review by anthonydahanne — APPROVED** [review]

**Pull request merged**